### PR TITLE
Use single progress bar

### PR DESCRIPTION
**Pull request by evgenyneu**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary

* Use a single progress bar instead of separate bars for warmup and sampling. This simplifies code since cmdstan output also combines warmup and sampling.

* Change the description of the bar from "warmup", to "sample" to "done" to indicate the stage of sampling.

* For Jupyter only, make each bar green as soon as it's done.

* I've deleted a lot of progress bar code following comment "read and process rest of the stdout if needed", I could not figure out what it was for, and everything worked for me without it.

#### In terminal

cmdstanpy_single_bar_terminal

#### In Jupyter

cmdstanpy_single_bar_jupyter

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): me

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

Yes, agree

## Code

To test the progress bar, run

```
python eight_schools.py
```

### eight_schools.py

```python
from cmdstanpy import CmdStanModel

model = CmdStanModel(stan_file="eight_schools.stan")

data = {
    "J": 8,
    "y": [28,  8, -3,  7, -1,  1, 18, 12],
    "sigma": [15, 10, 16, 11,  9, 11, 10, 18]
}

fit = model.sample(data=data, show_progress=True,
                   chains=4, cores=2,
                   sampling_iters=15000, warmup_iters=15000)

print(fit.summary())
```

### eight_schools.stan

```stan
data {
  int<lower=0> J;         // number of schools
  real y[J];              // estimated treatment effects
  real<lower=0> sigma[J]; // standard error of effect estimates
}
parameters {
  real mu;                // population treatment effect
  real<lower=0> tau;      // standard deviation in treatment effects
  vector[J] eta;          // unscaled deviation from mu by school
}
transformed parameters {
  vector[J] theta = mu + tau * eta;        // school treatment effects
}
model {
  target += normal_lpdf(eta | 0, 1);       // prior log-density
  target += normal_lpdf(y | theta, sigma); // log-likelihood
}
```

### CODE REVIEW

**Comment by ahartikainen** [comment]
A lot of the code was probably 'over-optimized'. The problem is that we only update everystep that Stan outputs update and if user sets that value to some weird cornercase then our bar might fail.

So I think it is safe to remove those bits and let's fix these issues when(if) they are seen.

File: cmdstanpy/model.py [hunk 1]
```diff
@@ -690,24 +676,25 @@ def sample(
                             dynamic_ncols = False
                         else:
                             dynamic_ncols = True
-                        pbar = [
-                            # warmup
-                            tqdm_pbar(
+
+                        # Total number of iterations (warmup + sample)
+                        # 2000 is default in cmdstan
+                        total_iters = 2000
```
Comment (ahartikainen) [inline cmdstanpy/model.py]: Will this work with non default values?

Comment (evgenyneu) [reply cmdstanpy/model.py]: Oops, it did work when both the number of warmup and sampling iterations were omitted by the user. But it did not work if only one of them was omitted.

Fixed here:
https://github.com/stan-dev/cmdstanpy/pull/197/commits/79169bc0e7c5230852749f9bf27bb9977c62ce01

Comment (evgenyneu) [reply cmdstanpy/model.py]: Nah, this is better

https://github.com/stan-dev/cmdstanpy/pull/197/commits/749ed90275081b99a98728ef8d82a5d726c63e44

Stan outputs 12 / 12345, so we know the total number of iterations is 12345. No need to guess that. This is how it worked before, I think.

**Comment by evgenyneu** [comment]
The build failed because `tqdm` Python library is not installed on build machine. Can we install it?

File: test/test_model.py [hunk 2]
```diff
@@ -3,9 +3,11 @@
 import os
 import shutil
 import unittest
+from unittest.mock import Mock
```
Comment (mitzimorris) [inline test/test_model.py]: does unittest.mock work with PyTest?

File: cmdstanpy/model.py [hunk 3]
```diff
@@ -1030,103 +1007,39 @@ def _read_progress(
...
-                                pbar_warmup.refresh()
-                                refresh_warmup = False
-                            # update values to full
-                            count, count_sampling = (
-                                raw_count - num_warmup - count_sampling,
-                                raw_count - num_warmup,
-                            )
-                            pbar_sampling.update(count)
+                        current_count = int(match.group(1))
+                        total_count = int(match.group(2))
```
Comment (mitzimorris) [inline cmdstanpy/model.py]: awesome cleanup!

File: test/test_model.py (hunk 2 above)
Comment (ahartikainen) [reply test/test_model.py]: Yes. Here is also one example from ArviZ https://github.com/arviz-devs/arviz/blob/master/arviz/tests/test_utils.py

File: test/test_model.py [hunk 4]
```diff
@@ -196,6 +198,36 @@ def test_model_compile_includes(self):
         model3 = CmdStanModel(stan_file=stan)
         self.assertTrue(model3.exe_file.endswith(exe.replace('\\', '/')))
 
+    def test_read_progress(self):
```
Comment (ahartikainen) [inline test/test_model.py]: Thanks for the test

**Comment by mitzimorris** [comment]
many thanks!

**Pull request merged**